File: LeetCode/Medium/Boats To Save People/Java/boatsToSavePeople.java
class Solution {
    public int numRescueBoats(int[] people, int limit) {
        Arrays.sort(people);
        int low=0, high = people.length-1;
        int cnt = 0;
        while(low <= high) {
            int total = people[low] + people[high];
            if (total <= limit) {
                low++;
                high--;
            }else {
                high--;
            }
            cnt++;
        }
        return cnt;
    }
}

File: LeetCode/Medium/Boats To Save People/c++/boats_to_save_people.cc
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) {
        int l = 0, h = people.size()-1;
        int sum = 0;
        int min_boats = 0;

        sort(people.begin(), people.end());

        while (l <= h) {
            sum = people[l] + people[h];
            if (sum <= limit) {
                l++;
            } 
            h--;
            min_boats++;
        }

        return min_boats;
    }
};
